core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -111,6 +111,25 @@ struct Rank {
   static constexpr Iterate inner_direction = InnerDir;
 };
 
+namespace Impl {
+// NOTE prefer C array U[M] to std::initalizer_list<U> so that the number of
+// elements can be deduced (https://stackoverflow.com/q/40241370)
+// NOTE commented out constexpr specifier below to work around a GCC bug that
+// was fixed from version 7.3
+// (https://gcc.gnu.org/bugzilla/show_bug.cgi?id=80935)
+template <class Array, class U, std::size_t M>
+/*constexpr*/ Array to_array_potentially_narrowing(const U (&init)[M]) {
```

Comment:
This would still not work to initialize a `constexpr` variable, see https://godbolt.org/z/vcj3zo.

Reply:
Your suggestion is fine but it still does not compile with GCC 5.3

Reply:
You can still declare the function `constexpr` with `gcc-5.3` but you can't use it in a `constexpr` context: https://godbolt.org/z/hPaef1 because `std::array::data()` is not `constexpr` before gcc-7.